x-pack/plugins/ml/public/application/datavisualizer/index_based/components/actions_panel/actions_panel.tsx

```diff
@@ -31,7 +31,7 @@ export const ActionsPanel: FC<Props> = ({ indexPattern }) => {
   function openAdvancedJobWizard() {
     // TODO - pass the search string to the advanced job page as well as the index pattern
     //       (add in with new advanced job wizard?)
-    window.open(`#/jobs/new_job/advanced?index=${indexPattern}`, '_self');
+    window.open(`#/jobs/new_job/advanced?index=${indexPattern.id}`, '_self');
```

Comment:
This change is causing the link to the advanced job editor in the data visualizer to fail.

![image](https://user-images.githubusercontent.com/7405507/91969820-a57d5180-ed0e-11ea-85e3-bdd599888734.png)

Your change seems to have exposed an issue with the routing code in the ML plugin. In order to fix the link to the job wizard and get this PR through, I suggest making a slight change to the way the URL is built up here.

Add this to the list of imports:
```
import { getBasePath } from '../../../../util/dependency_cache';
```

and then construct the URL by:

```
const basePath = getBasePath();
window.open(`${basePath.get()}/app/ml/jobs/new_job/advanced?index=${indexPattern.id}`, '_self');
```

This fixes the link in the ML data visualizer for me (hopefully this will fix the associated functional tests too).

We can then follow-up with a cleaner refactor as part of the work we are already looking at as part of our switch to BrowserRouter in https://github.com/elastic/kibana/issues/72013.